core/templates/dev/head/pages/landing/LandingPage.js

```diff
@@ -0,0 +1,167 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controller for landing page.
+ */
+
+oppia.constant('LANDING_PAGE_DATA', {
+  maths: {
+    fractions: {
+      collection_id: '4UgTQUc1tala',
+      student: {
+        row_1: {
+          heading: 'Fractions just got easier.',
+          image: 'hero_bakery_combined.png'
+        },
+        row_2: {
+          heading: 'Fractions don’t have to be hard',
+          paragraph_1: 'Learning about fractions can be tricky. With so ' +
+            'much to understand, it’s easy to get confused with the concepts.',
+          paragraph_2: 'That’s where we come in! Oppia makes it easy to ' +
+            'quickly jump in and learn the fundamentals of fractions.',
+          image: 'intro_matthew.svg',
+        },
+        row_3: {
+          heading: 'Fun storytelling for all',
+          paragraph_1: 'Every fractions lesson follows our friend, ' +
+            'Matthew, as he works hard to become a great baker!',
+          paragraph_2: 'We’ll make sure you’re not confused along the way ' +
+            'with helpful hints at every step.',
+          image: 'storytelling_combined.png'
+        },
+        row_4: {
+          heading: 'Easy to follow lessons',
+          paragraph_1: 'With Oppia, we make it easy to go through the ' +
+            'lessons and learn something new!',
+          paragraph_2: 'You will work to learn new concepts and help ' +
+            'Matthew become a great baker along the way. We make it fun and ' +
+            'easy to get started.',
+          image: 'lessons_lessons.svg'
+        }
+      },
+      teacher: {
+        row_1: {
+          heading: 'Fractions just got easier',
+          image: 'matthew_paper.png'
+        },
+        row_2: {
+          heading: 'Fun storytelling for all',
+          paragraph_1: 'Students are guided through explorations with ' +
+            'targeted feedback and immersive storytelling.',
+          paragraph_2: 'Oppia guides students step-by-step with helpful ' +
+            'hints, so they can complete the lessons on their own.',
+          image: 'matthew_fractions.png',
+        },
+        row_3: {
+          heading: 'Easy-to-follow lessons',
+          paragraph_1: 'By working through lessons on Oppia, your young ' +
+            'learners can apply their knowledge to real-world problems.',
+          paragraph_2: 'Our lessons also have audio subtitles, to support ' +
+            'students with reading difficulties.',
+          video: 'fractions_video.mp4',
+        }
+      }
+    }
+  }
+});
+
+oppia.controller('Fractions', [
+  '$scope', '$timeout', '$window', 'SiteAnalyticsService',
+  'UrlInterpolationService', 'LANDING_PAGE_DATA', function(
+      $scope, $timeout, $window, SiteAnalyticsService,
+      UrlInterpolationService, LANDING_PAGE_DATA) {
+    var url = new URL($window.location.href);
```

Comment:
This stuff should be part of some kind of URL analyzer service. I think we have that somewhere, can we use it instead?